src/Classes/TradeQueryGenerator.lua

```diff
@@ -0,0 +1,695 @@
+for i, curInfluenceInfo in ipairs(itemLib.influenceInfo) do
+    influenceDropdownNames[i + 1] = curInfluenceInfo.display
+    hasInfluenceModIds[i] = "pseudo.pseudo_has_" .. string.lower(curInfluenceInfo.display) .. "_influence"
+end
+
+
+-- This is not a complete list as most mods get caught with the find "Local" check, or don't overlap with non-local mods so we don't care. Some groups
+-- are also shared between local and non-local mods, so a group only approach is not viable.
+local localOnlyModGroups = {
+    ["BaseLocalDefences"] = true,
+    ["BaseLocalDefencesAndLife"] = true,
+    ["LocalIncreasedPhysicalDamagePercentAndAccuracyRating"] = true,
+    ["LocalPhysicalDamagePercent"] = true,
+    ["DefencesPercent"] = true,
+    ["DefencesPercentAndStunRecovery"] = true,
+    ["LocalAttributeRequirements"] = true,
+    ["DefencesPercentSuffix"] = true
+}
+
+local MAX_FILTERS = 36
+
+local function logToFile(...)
+    --ConPrintf(...)
+    --local f = io.open('../TradeQueryLog.log', 'a')
+    --f:write(string.format(...))
+    --f:write("\n")
+    --f:close()
+end
+
+local TradeQueryGeneratorClass = newClass("TradeQueryGenerator", function(self, queryTab)
+    self:InitMods()
+    self.queryTab = queryTab
+    self.itemsTab = queryTab.itemsTab
+    self.calcContext = { }
+
+    table.insert(main.onFrameFuncs, function()
+        self:OnFrame()
+    end)
+end)
+
+local function fetchStats()
+    local tradeStats = ""
+	local easy = common.curl.easy()
+	easy:setopt_url("https://www.pathofexile.com/api/trade/data/stats")
+    easy:setopt_useragent("Path of Building/" .. launch.versionNumber)
+	easy:setopt_writefunction(function(data)
+		tradeStats = tradeStats..data
+		return true
+	end)
+	easy:perform()
+	easy:close()
+    return tradeStats
+end
+
+local function stripInfluenceSuffix(key)
+    local influenceSuffixPos = nil
+    for _, suffix in ipairs(influenceSuffixes) do
+        influenceSuffixPos = key:find(suffix)
+        if influenceSuffixPos ~= nil then
+            return key:sub(1, influenceSuffixPos - 1)
+        end
+    end
+    return key
+end
+
+local function canModSpawnForItemCategory(mod, tags)
+    for i, key in ipairs(mod.weightKey) do
+        local influenceStrippedKey = stripInfluenceSuffix(key)
+        if key ~= "default" and mod.affix:find("Elevated") ~= nil and tags[influenceStrippedKey] == true then
+            return true
+        elseif key ~= "default" and mod.type == "Corrupted" and tags[influenceStrippedKey] == true then
+            return true
+        elseif mod.weightVal[i] > 0 and tags[influenceStrippedKey] == true then
+            return true
+        end
+    end
+    return false
+end
+
+function TradeQueryGeneratorClass:GenerateModData(mods, tradeQueryStatsParsed)
+    for modId, mod in pairs(mods) do
+        if localOnlyModGroups[mod.group] == true or (modId:find("Local") ~= nil and modId:find("Socketed") == nil) then -- skip all local mods other than socket level mods
```

Comment:
You might be able to parse locality from the ggpk.  I just don't think we save it off into ModItem.lua